docs/guestfs.md

```diff
@@ -0,0 +1,165 @@
+
+[Libguestfs tools](https://libguestfs.org/) are a set of utilities for accessing and modifying VM disk images. The command `virtclt guestfs` helps to deploy an interactive container with the libguestfs-tools and the PVC attached to it. This command is particurarly useful if the users need to modify, inspect or debug VM disks on a PVC.
+```bash
+$ virtctl guestfs -h
+Create a pod with libguestfs-tools, mount the pvc and attach a shell to it. The pvc is mounted under the /disks directory inside the pod for filesystem-based pvcs, or as /dev/vda for block-based pvcs
+
+Usage:
+  virtctl guestfs [flags]
+
+Examples:
+  # Create a pod with libguestfs-tools, mount the pvc and attach a shell to it:
+  virtctl guestfs <pvc-name>
+
+Flags:
+  -h, --help                 help for guestfs
+      --image string         libguestfs-tools container image
+      --kvm                  Use kvm for the libguestfs-tools container (default true)
+      --pull-policy string   pull policy for the libguestfs image (default "IfNotPresent")
+
+Use "virtctl options" for a list of global command-line options (applies to all commands).
+```
+
+The `virctl guestfs` sets up by default `kvm` for the interactive container. This considerably speeds up the execution of the libguestfs-tools since they use QEMU. If the cluster doesn't have any kvm supporting nodes, the user must disable kvm by setting the option `--kvm=false`. If not set, the libguestfs-tools pod will remain pending because it cannot be scheduled on any node.
+
+The command uses by default the image exposed by KubeVirt under the http endpoint `/apis/subresources.kubevirt.io/<kubevirt-version>/guestfs` but it can be configured to use a custom image by using the option `--image`. The users can also overwrite the pull policy of the image by setting the option `pull-policy`.
+
+Currently, the `virtctl guestfs` accepts only a single PVC and in the future might be extended to support multiple PVCs attached to the interactive pod.
+
+## Examples and use-cases
+Generally, the user can take advantage of the `virtctl guestfs` command for all the typical usage of libguestfs-tools and it is strongly recommended to consult the [official documentation](https://libguestfs.org/guestfs-recipes.1.html). This command simply aims to help in configuring the correct containerized environment in the Kubernetes cluster where KubeVirt is installed.
+
+For all the examples, the user has to start the interactive container by referencing the PVC to the `virtctl guestfs` command. This will deploy the interactive pod and attach the stdin and stdout. 
+
+Example:
+
+```bash
+$ virtctl guestfs pvc-test
+Use image: registry:5000/kubevirt/libguestfs-tools@sha256:6644792751b2ba9442e06475a809448b37d02d1937dbd15ad8da4d424b5c87dd 
+The PVC has been mounted at /disk 
+Waiting for container libguestfs still in pending, reason: ContainerCreating, message:  
+Waiting for container libguestfs still in pending, reason: ContainerCreating, message:  
```

Comment:
Not for the user guide, but for the container:

What about a small "motd" file to display the user the available commands when the user is getting into the shell?

Reply:
yes, we can maybe add it to the help function?